word_counter.rb

```diff
@@ -0,0 +1,18 @@
+class WordCounter
+  def self.count file  
+    raise 'WordCounter: no file given' unless file 
+    # use inject to create a hash of word counts
+    to_word_array(file).inject({}) { |h,w| h[w] ? h[w] += 1 : h[w] = 1; h }
```

Comment:
This is a bit hard to follow with all these methods chained together - maybe split them out?  Also, more illustrative variable names (than `h` and `v`) will improve clarity.